File: frontend/src/shared/models/dtos/token-dto.ts
import { Token } from '@models';

export interface TokenDto {
    id?: string; // UUID
    appointmentId?: number | null;
    categoryId: number;
    userId: string; // UUID
    purchasedAt?: Date;
    redeemedAt?: Date;
}

export const fromToken = (token: Token): TokenDto => {
    return {
        id: token.id,
        appointmentId: token.appointmentId,
        categoryId: token.categoryId,
        userId: token.userId,
        purchasedAt: token.purchasedAt,
        redeemedAt: token.redeemedAt,
    };
};

File: backend/src/main/java/at/htl/courseschedule/dto/TokenDto.java
package at.htl.courseschedule.dto;

import at.htl.courseschedule.entity.Token;

import java.util.UUID;

public record TokenDto(UUID id, Long appointmentId, Long categoryId, UUID userId) {
    public static TokenDto fromToken(Token token) {
        return new TokenDto(
                token.getId(),
                token.getAppointment() != null ? token.getAppointment().getId() : null,
                token.getCategory().getId(),
                token.getUser().getId()
        );
    }
}
